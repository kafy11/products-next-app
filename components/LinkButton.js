import LinkNext from './LinkNext';

//A bootstrap link button
const LinkButton = ({ children, title, href, className }) => (
  <LinkNext 
    href={href}
    role="button"
    className={`btn btn-link ${className}`} 
    title={title}
  >
    {children}
  </LinkNext>
);

export default LinkButton;