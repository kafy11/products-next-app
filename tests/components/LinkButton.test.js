import { shallow } from 'enzyme';
import LinkButton from '../../components/LinkButton';

let wrapper;
const props = {
    children: "Clique",
    href: "localhost:3000",
    className: "testClass",
    title: "Clique aqui",
};

beforeEach(() => {
    wrapper = shallow(<LinkButton {...props} />);
});

describe('LinkButton', () => {
    it('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
    it('should render the children correctly', () => {
        const children = wrapper.find('LinkNext').prop('children');
        expect(children).toBe(props.children);
    });
    
    it('should render the href correctly', () => {
        const href = wrapper.find('LinkNext').prop('href');
        expect(href).toBe(props.href);
    });
    
    it('should render the title correctly', () => {
        const title = wrapper.find('LinkNext').prop('title');
        expect(title).toBe(props.title);
    });
    
    it('should add the className', () => {
        const hasClass = wrapper.find('LinkNext').hasClass(props.className);
        expect(hasClass).toBe(true);
    });
});