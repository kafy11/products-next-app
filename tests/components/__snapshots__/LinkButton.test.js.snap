// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkButton should render correctly 1`] = `
<LinkNext
  className="btn btn-link testClass"
  href="localhost:3000"
  role="button"
  title="Clique aqui"
>
  Clique
</LinkNext>
`;
